xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1402,6 +1402,48 @@ namespace VIDEO
       if (!strTrailer.empty())
         movieDetails.m_strTrailer = strTrailer;
 
+      // ** Deal with 'Disc' subdirectories and add to/create a set
+      if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+              CSettings::SETTING_MYVIDEOS_HANDLEDISCN))
+      {
+        size_t iDisc = pItem->GetDiscPath(movieDetails.m_strFileNameAndPath);
+        if (iDisc)
+        {
+          if (movieDetails.m_set.title.empty())
+          {
+            // Get existing sets
+            CFileItemList listItems;
+            std::string baseDir = "videodb://movies/sets/?ignoreSingleMovieSets=false";
+            if (CDirectory::GetDirectory(baseDir, listItems, "", DIR_FLAG_DEFAULTS))
+            {
+              // See if set exists
+              for (int listIndex = 0; listIndex < listItems.Size(); listIndex++)
```

Comment:
```
for (const auto& item :: listItems)
{
  const std::string currentSet{m_database.GetSetById(item->GetVideoInfoTag()->m_iDbId)};
  ...
}
```

Reply:
Can I just check it should be `for (const auto& item : listItems)` (with one :) - I'm not hugely familiar with C++ and haven't come accross range based loops before.

Reply:
Yes, that was a typo on my end, sorry. One `:`, please.